nemo_curator/classifiers/prompt_task_complexity.py

```diff
@@ -0,0 +1,369 @@
+        batch_size (int): The number of samples per batch for inference. Defaults to 256.
+        text_field (str): The field in the dataset that should be classified.
+        max_chars (int): The maximum number of characters in each document to consider for classification. Defaults to 2000.
+        device_type (str): The type of device to use for inference, either "cuda" or "cpu". Defaults to "cuda".
+        autocast (bool): Whether to use mixed precision for faster inference. Defaults to True.
+        max_mem_gb (int, optional): The maximum amount of memory in GB to allocate for the model. If None,
+                                    it defaults to the available GPU memory minus 4 GB.
+
+    """
+
+    def __init__(
+        self,
+        batch_size: int = 256,
+        text_field: str = "text",
+        max_chars: int = 2000,
+        device_type: str = "cuda",
+        autocast: bool = True,
+        max_mem_gb: Optional[int] = None,
+    ):
+        self.text_field = text_field
+
+        config = AutoConfig.from_pretrained(PROMPT_TASK_COMPLEXITY_IDENTIFIER)
+        pred_column = config.targets
+
+        model_config = PromptTaskComplexityConfig(
+            model_output_type=config.model_output_type
+        )
+        model = PromptTaskComplexityModel(
+            config=model_config, autocast=autocast, max_mem_gb=max_mem_gb
+        )
+
+        super().__init__(
+            model=model,
+            labels=None,
+            filter_by=None,
+            batch_size=batch_size,
+            out_dim=None,
+            pred_column=pred_column,
+            max_chars=max_chars,
+            device_type=device_type,
+            autocast=autocast,
+        )
+
+    def _run_classifier(self, dataset: DocumentDataset) -> DocumentDataset:
+        print("Starting prompt task and complexity classifier inference", flush=True)
+
+        df = dataset.df
+        columns_to_keep_list = df.columns.to_list()
+        df["sliced_text"] = df[self.text_field].str.slice(0, self.max_chars)
+
+        model = self.model
+        classifier_pipe = op.Sequential(
+            op.Tokenizer(model, cols=["sliced_text"], tokenizer_type="default"),
+            op.Predictor(
+                model,
+                sorted_data_loader=True,
+                batch_size=self.batch_size,
+                model_output_cols=self.pred_column,
+            ),
+            repartition=df.npartitions,
+            keep_cols=columns_to_keep_list,
+        )
+
+        df = classifier_pipe(df)
+        return DocumentDataset(df)
+
+    def _filter_documents(
+        self,
+        dataset: DocumentDataset,
+    ) -> DocumentDataset:
+        raise NotImplementedError(
+            "filter_by not supported with PromptTaskComplexityClassifier"
+        )
+
+    def get_labels(self):
+        print(
+            "Please see https://huggingface.co/nvidia/prompt-task-and-complexity-classifier for more information about PromptTaskComplexityClassifier outputs."
```

Comment:
I dont follow this, what are expected labels here ? 

And why are we returning None ?

Reply:
Since there are 11 output columns I wasn't really sure if we should support this function. Although I guess we could have it return a string or dictionary about the outputs? What do you think?

Reply:
Dont think we should support it. You should raise a NotImplementedError here with the error message.